File: python/stack.py
class Stack:
    def __init__(self):
        self._data = []

    def push(self, val: int):
        self._data.append(val)

    def pop(self):
        return self._data.pop() if self._data else None

    def top(self):
        return self._data[-1] if self._data else None

    def size(self) -> int:
        return len(self._data)

    def is_empty(self) -> bool:
        return len(self._data) == 0

    def clear(self):
        self._data.clear()

File: src/stack.cpp
#include "stack.h"
void Stack::push(int val) { data.push_back(val); }
void Stack::pop() { if (!data.empty()) data.pop_back(); }
int Stack::top() const { return data.empty() ? 0 : data.back(); }
void Stack::print() const {
    for (size_t i = 0; i < data.size(); ++i) {
        std::cout << data[i] << (i == data.size() - 1 ? "" : " ");
    }
    std::cout << "\n";
}
int Stack::size() const { return data.size(); }
void Stack::clear() { data.clear(); }
bool Stack::empty() const { return data.empty(); }